packages/compartment-mapper/src/types.d.ts

```diff
@@ -0,0 +1,11 @@
+// ESlint does not understand export *
+/* eslint-disable import/export */
+
+export * from './types/external.js';
```

Comment:
Since you have this indirection, all the modules below can be `.ts`. Moreover they should be, because they are pure types rather than annotations about other `.js` files.

It would also prevent ignoring them when `skipLibCheck: true`

Reply:
For the record, we reversed this decision out of band. We are enforcing the limit with the build configuration and I’ve renamed these `.ts` so that `tsc` will complain about errors in them, per yours and @boneskull’s observation. The barrel `types.d.ts` (external) and `src/types.d.ts` (internal)  remain in `.d.ts`.